root/programs/C#/Frameworks/Infrastructure/Framework/Util/GetMessage.cs

```diff
@@ -119,307 +107,179 @@ public static string GetMessageDescription(string messageID)
-                }
-                else if (defaultXMLFilePath_ResourceLoaderExists
-                    || cultureWiseXMLFilePath_ResourceLoaderExists)
-                {
-                    // Load normally.
-                    xMLMSG.Load(PubCmnFunction.BuiltStringIntoEnvironmentVariable(tempXMLFilePath));
-
-                    //added by ritu
-                    GetMessage.FillDictionary(xMLMSG, innerDictionary);
-                    // and initialize DicMSG[tempXMLFileName]
-                    DicMSG[tempXMLFileName] = innerDictionary;
-                }
-                else
-                {
-                    // If the file is not found, initialized as empty XML
-                    xMLMSG.LoadXml("<?xml version=\"1.0\" encoding=\"utf-8\" ?><MSGD></MSGD>");
-
-                    //added by ritu
-                    GetMessage.FillDictionary(xMLMSG, innerDictionary);
-
-                    // and initialize DicMSG[tempXMLFileName]
-                    DicMSG[tempXMLFileName] = innerDictionary;
-                    //xmlfilenotExist = true;
-                }
-
-                // and return GetMessageDescription.
-                if (DicMSG[tempXMLFileName].ContainsKey(messageID))
-                {
-                    return GetMessage.DicMSG[tempXMLFileName][messageID];
-                }
-                else
-                {
-                    return string.Empty;
-                }
-
-                #endregion
             }
+
+            return content;
         }
 
-        /// <summary>Get culture wise XML file name</summary>
+        /// <summary>Get culture wise file</summary>
         /// <param name="cultureName">cultureName</param>
-        /// <param name="defaultXMLFilePath">defaultXMLFilePath</param>
-        /// <param name="defaultXMLFileName">defaultXMLFileName</param>
-        /// <param name="cultureWiseXMLFilePath">cultureWiseXMLFilePath</param>
-        /// <param name="cultureWiseXMLFileName">cultureWiseXMLFileName</param>
-        private static void GetCultureWiseXMLFileName(
+        /// <param name="defaultFilePath">defaultFilePath</param>
+        /// <param name="cultureWiseFilePath">cultureWiseFilePath</param>
+        /// <param name="cultureWiseFileName">cultureWiseFileName</param>
+        /// <returns>defaultFileName</returns>
+        private static string GetCultureWiseFile(
             string cultureName,
-            string defaultXMLFilePath,
-            out string defaultXMLFileName,
-            out string cultureWiseXMLFilePath,
-            out string cultureWiseXMLFileName
+            string defaultFilePath,
+            out string cultureWiseFilePath,
+            out string cultureWiseFileName
             )
         {
-            defaultXMLFileName = Path.GetFileName(defaultXMLFilePath);
+            string defaultFileName = Path.GetFileName(defaultFilePath);
+            int temp = defaultFileName.Length - 4; // -4 は *.xxx の拡張子部分
```

Comment:
@daisukenishino 

Is the length of the extension always four?
It might be a good idea to use `Path.GetFileNameWithoutExtension` method.

```cs
int temp = Path.GetFileNameWithoutExtension(defaultFilePath).Length;
```

or

```cs
cultureWiseFileName = Path.GetFileNameWithoutExtension(defaultFilePath) + "_" + Path.GetExtension(defaultFilePath)
```